platform/api/aws/images.go

```diff
@@ -585,19 +586,75 @@ func (a *API) copyImageIn(sourceRegion, sourceImageID, name, description string,
 	// constraint that multiple images cannot have the same name.
 	// As a result we could have created a duplicate image after
 	// losing a race with a CopyImage task created by a previous run.
-	// Don't try to clean this up automatically for now, but at least
-	// detect it so plume pre-release doesn't leave any surprises for
-	// plume release.
-	_, err = a.FindImage(name)
+	// Make an attempt to clean this up automatically now.
+	dedupedID, err := a.findAndDedupeImage(name)
 	if err != nil {
-		return "", fmt.Errorf("checking for duplicate images: %v", err)
+		return "", fmt.Errorf("error checking for duplicate images: %v", err)
+	}
+	if dedupedID == "" {
+		return "", fmt.Errorf("did not find image %q", name)
+	}
+
+	return dedupedID, nil
+}
+
+// findAndDedupeImage finds an image given its name, but also attempts to
+// automatically handle duplicate private images. If multiple images, all
+// private, are found, it deregisters all but one.
+// This is necessary because, although names are meant to be unique, use of the
+// "CopyImage" API semi-frequently results in duplicates.
+// If the name doesn't refer to any images, an empty string and no error will
+// be returned.
+func (a *API) findAndDedupeImage(name string) (string, error) {
+	images, err := a.findImages(name)
+	if err != nil || len(images) == 0 {
+		return "", err
+	}
+	if len(images) == 1 {
+		return *images[0].ImageId, nil
+	}
+
+	for _, image := range images {
+		if *image.Public {
+			return "", fmt.Errorf("multiple images found named %q; could not automatically resolve because %q is already public", name, *image.ImageId)
+		}
+	}
+
+	ids := mapImagesToIds(images)
+	// Deterministically pick an id to keep
+	sort.Strings(ids)
+
+	plog.Infof("found multiple images named %q (%v): keeping %q", name, ids, ids[0])
+
+	for i := 1; i < len(ids); i++ {
+		_, err := a.ec2.DeregisterImage(&ec2.DeregisterImageInput{
+			ImageId: aws.String(ids[i]),
+		})
+		if err != nil {
```

Comment:
In practice, this does happen. From my attempt to test this in our dev account:

```
error trying to cleanup dupe images named "Container-Linux-developer-1702.1.0_jenkins2-build-1702_local-1246-hvm" ([ami-5a7de525 ami-e67be399]): InvalidAMIID.Unavailable: The image ID 'ami-e67be399' is no longer available
```

It looks like this error will have to be ignored since that happens pretty easily.